contracts/kernel/organs/AccountingApp.sol

```diff
@@ -0,0 +1,98 @@
+pragma solidity ^0.4.8;                                                         
+
+contract AccountingApp {
+    
+    // The concept of sending tokens from one address to another
+    struct Transaction {
+        address token;
+        int value;
+        address baseToken;
+        address baseValue;
+        
+        address from;
+        address to;
+        string reference;
+
+        uint period;
+        // Back reference the states?
```

Comment:
We have to see what is cheaper gas wise. Having a `mapping (uint => TransactionAction)` could turn out to be cheaper